pkg/lakom/verifysignature/verifier.go

```diff
@@ -87,7 +87,7 @@ func verify(ctx context.Context, imageRef name.Reference, keys []crypto.PublicKe
 			}
 
 			if IsNoMatchingSignature(err) {
-				if errors.Is(err, context.Canceled) {
+				if errors.Is(ctx.Err(), context.Canceled) || errors.Is(ctx.Err(), context.DeadlineExceeded) {
```

Comment:
Can you add log messages to make it easier understand this issue if it occur again or if we have some other issues around this code, for example:

```suggestion
				if errors.Is(ctx.Err(), context.Canceled) || errors.Is(ctx.Err(), context.DeadlineExceeded) {
					log.Info("no matching signatures error detected as canceled or deadline exceeded context", "error", err)
```

But you can add more where you think is necessary.